### ISSUE TITLE
Is the pytorch backend correctly disabling gradient calculation?

### ISSUE THREAD

**Issue by jjancsary** [triton-inference-server/server#2491]
**Description**
In my understanding, when running the forward pass of a TorchScript model, unless further action is taken, PyTorch tracks all the information needed to compute the gradient with respect to the model parameters. This can be costly, both in terms of CPU time, and in terms of memory. For some of our models we observe a factor 5x increase in RSS if gradient calculation is enabled.

In Python, to disable gradient calculation, a context manager invoking torch.no_grad() is needed:
https://pytorch.org/docs/stable/generated/torch.no_grad.html

In C++, there's a similar construct via a scope guard (torch::NoGradGuard):
https://pytorch.org/cppdocs/api/typedef_namespacetorch_1abf2c764801b507b6a105664a2406a410.html

See also bug ticket https://github.com/pytorch/pytorch/issues/17095.

I checked the code of the Triton pytorch_backend, and I don't see torch::NoGradGuard used anywhere. Empirically, we observe very high memory usage when using TorchScript models in Triton; similarly high to when we don't use torch.no_grad() in Python.

Are you disabling gradient calculation via some other means I'm unaware of?

**Triton Information**
What version of Triton are you using?
20.10, but checked the main branch as well.

Are you using the Triton container or did you build it yourself?
We're using pre-built 20.10.

**To Reproduce**
Steps to reproduce the behavior.
Use any TorchScript model in Triton.

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).
Hopefully not needed, as any TorchScript model should allow to reproduce the issue.

**Expected behavior**
Similarly low memory usage in Triton to when we use torch.no_grad() in Python.

**Comment by jjancsary** [comment]
Probably related: https://github.com/triton-inference-server/server/issues/2398

**Comment by CoderHam** [comment]
Thank you @jjancsary for pointing this out. We were not disabling gradient calculation.

**Comment by CoderHam** [comment]
Fixed by https://github.com/triton-inference-server/pytorch_backend/pull/12
